### ISSUE TITLE
Code quality in srchook/ntsdlfn.c

### ISSUE THREAD

**Issue by kawapure** [freedom7341/NTStyle#4]
Hello, I was looking through the code in `srchook/ntsdlfn.c`, which temporarily loads a few DWM API functions in during runtime, and have a few questions about the existing design. Notably, I believe that the existing code has a tiny resource leak in successful instances.

Current code is structured something like this for both function imports:

```c
HMODULE hLib = LoadLibrary(L"Dwmapi.dll");
HRESULT hRet = 0;

if (hLib)
{
	FARPROC fLib = GetProcAddress(hLib, <FUNCTION>);
	if (hRet = (HRESULT)fLib(<ARGUMENTS>))
	{
		FreeLibrary(hLib);
		return hRet;
	}
}

return 0;
```

I believe that the `FreeLibrary` call should be ran in both cases. Otherwise, what would happen here is that the library is loaded by the application and never freed. Note that S_OK (a successful HRESULT) is equal to 0, so the if condition is passed and the function always returns successful if `hRet` is 0, without freeing the library. In general, the macros `SUCCEEDED()` or `FAILED()` should be used for HRESULTs, for clarity.

If this is the intended behaviour, then it'd be good to add a comment documenting this. Otherwise, appropriate adjustments should be made.

**Comment by freedom7341** [comment]
Makes sense. I'll restructure these functions in the next commit. Thank you for taking a look at this!

**Comment by freedom7341** [comment]
Should be fixed in commit [a9d580d](https://github.com/Vortesys/NTStyle/commit/a9d580d8b960c720cf68d83b642a29461749721b).
